### PR TITLE
Eliminate excessive error reporting in some cases.

### LINKED ISSUE #60: Excessive error reporting  when a trait isn't found in scope.
This is clearly overkill, I mean, c'mon, I _get it_. 

Screenshot from 2021-05-28 15-48-59

Clearly this should only be one error instead of two.

### PR DESCRIPTION
**Pull request by emilyaherbert**
Closes #60 . I tried to reproduce the provided issue as best as I could. If you can think of any more possible cases that you remember seeing / think may be an issue, let me know and I can add them to this PR.

### CODE REVIEW

**Review by nfurfaro — DISMISSED** [review]
Looks good. Will be nice to see this merged!

File: test/src/e2e_vm_tests/test_programs/excessive_error_reporting/Forc.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[project]
+author = "Fuel Labs <contact@fuel.sh>"
+license = "Apache-2.0"
+name = "excessive_error_reporting"
+entry = "main.sw"
+
+
+[dependencies]
+std = { git = "http://github.com/FuelLabs/sway-lib-std", version = "v0.0.1" }
+core = { git = "http://github.com/FuelLabs/sway-lib-core", version = "v0.0.1" }
```
Comment (sezna) [inline test/src/e2e_vm_tests/test_programs/excessive_error_reporting/Forc.toml:10]:
```suggestion
```

File: sway-core/src/semantic_analysis/ast_node/impl_trait.rs [hunk 2]
```diff
@@ -45,9 +45,11 @@ pub(crate) fn implementation_of_trait(
             type_arguments[0].clone().name_ident.span().clone(),
         ));
     }
+    let mut prob_warnings = vec![];
+    let mut prob_errors = vec![];
     match namespace
         .get_call_path(&trait_name)
-        .ok(&mut warnings, &mut errors)
+        .ok(&mut prob_warnings, &mut prob_errors)
```
Comment (sezna) [inline sway-core/src/semantic_analysis/ast_node/impl_trait.rs:52]: this just throws away the errors from `get_call_path`?

**Review by sezna — CHANGES_REQUESTED** [review]
I'm not sure what this PR is trying to do. It looks like it is throwing away any "not found" errors that would come from the namespace.

Additionally, and this is my fault for not going through the issues, #60 is actually not really an issue any more. Back then we still had the brute force multi-pass compilation, and it would continually append the same error for each iteration. That's what caused #60.

**Comment by emilyaherbert** [comment]
I think there was a miscommunication about what the issue was about. I thought that #60 was asking for the two distinct errors to turn into one distinct error. i.e. to remove the "symbol not found" error because that information is basically repeated in the second error.

**Pull request closed without merging**